src/Aspire.Hosting/Dashboard/ResourceLogSource.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Channels;
+using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Dcp.Model;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal sealed class ResourceLogSource<R>(IKubernetesService kubernetesService, R resource):
+    IAsyncEnumerable<IReadOnlyList<(string Content, bool IsErrorMessage)>>
```

Comment:
Does C# have anything like TypeScript's type definitions? `IAsyncEnumerator<IReadOnlyList<(string Content, bool IsErrorMessage)>>` is repeated a lot of times.